src/common/config.h

```diff
@@ -136,6 +212,8 @@ struct md_config_t {
   // No metavariables will be returned (they will have already been expanded)
   int get_val(const char *key, char **buf, int len) const;
   int _get_val(const char *key, char **buf, int len) const;
+  config_value_t get_val(const char *key) const;
```

Comment:
I've added a template method:

template<typename T> T get_val(const char *key) const;

However checking for an error is still necessary (say, if there's no such an option, or it has a different type)